src/main/java/org/cyclonedx/maven/BaseCycloneDxMojo.java

```diff
@@ -680,28 +707,104 @@ private Component.Type resolveProjectType() {
+                    }
+                }
+            };
             rootNode.accept(visitor);
             for (final DependencyNode dependencyNode : visitor.getNodes()) {
-                buildDependencyGraphNode(componentRefs, dependencies, dependencyNode, null);
+                buildDependencyGraphNode(dependencies, dependencyNode, null, excludedNodes, resolvedPUrls, loggedReplacementPUrls);
+            }
+
+            /*
+             Test and Runtime scope artifacts may conceal transitive compile dependencies.
+             Test artifacts will conceal other artifacts if includeTestScope is false, whereas Runtime artifacs
+             will conceal other artifacts if both includeTestScope and includeRuntimeScope are false.
+             */
+            if ((includeCompileScope && !includeTestScope) || !excludedNodes.isEmpty()){
+                final ProjectBuildingRequest testBuildingRequest = getProjectBuildingRequest(mavenProject);
+                final Map<String, DependencyNode> concealedNodes = new HashMap<>();
+                final Map<String, DependencyNode> concealedEmptyNodes = new HashMap<>();
+                if (includeCompileScope && !includeTestScope) {
+                    final DependencyNode testRootNode = dependencyCollectorBuilder.collectDependencyGraph(testBuildingRequest, null);
+                    for (DependencyNode child: testRootNode.getChildren()) {
+                        if (Artifact.SCOPE_TEST.equals(child.getArtifact().getScope())) {
+                            collectNodes(concealedNodes, concealedEmptyNodes, child);
+                        }
+                    }
+                    if (!includeRuntimeScope) {
+                        collectRuntimeNodes(concealedNodes, concealedEmptyNodes, testRootNode);
+                    }
+                }
+                if (!excludedNodes.isEmpty()) {
+                    for (DependencyNode excluded: excludedNodes.values()) {
+                        collectNodes(concealedNodes, concealedEmptyNodes, excluded);
+                    }
+                }
+
+                final Set<String> loggedPurls = new HashSet<>();
+                final Deque<Dependency> toProcess = new ArrayDeque<>(dependencies.values());
+                while (!toProcess.isEmpty()) {
+                    final Dependency dependency = toProcess.remove();
+                    if ((dependency.getDependencies() == null) || dependency.getDependencies().isEmpty()) {
+                        final String purl = dependency.getRef();
+                        DependencyNode concealedNode = concealedNodes.get(purl);
+                        if (concealedNode == null) {
+                            concealedNode = concealedEmptyNodes.get(purl);
+                        }
+                        if (concealedNode != null) {
+                            if (!loggedPurls.contains(purl)) {
+                                getLog().info("CycloneDX: Populating concealed node: " + purl);
```

Comment:
mix of previous comments: INFO is too chatty, and "concealed" is hard to understand

Reply:
Sounds good, will take care of this today

Reply:
New PR pushed to address these changes